validphys2/src/validphys/pseudodata.py

```diff
@@ -236,7 +236,132 @@ def indexed_make_replica(groups_index, make_replica):
 
     return pd.DataFrame(make_replica, index=groups_index, columns=["data"])
 
+def make_level0_data(data,fakepdf):
+    """
+    GENERAL DESCRIPTION:
+
+    Given a validphys.core.DataGroupSpec object, load commondata and 
+    generate a new commondata instance with central values replaced
+    by fakepdf prediction 
+
+    Parameters
+    ----------
+
+    data : validphys.core.DataGroupSpec 
+    
+    fakepdf: validphys.core.PDF
+
+    Returns
+    -------
+    list
+        list of validphys.coredata.CommonData instances corresponding to
+        all datasets within one experiment. The central value is replaced 
+        by Level 0 fake data.
+
+    Example
+    -------
+    >>> from validphys.api import API
+    >>> API.make_level0_data(dataset_inputs = [{"dataset":"NMC"}], use_cuts="internal", theoryid=200,fakepdf = "NNPDF40_nnlo_as_01180")
+    
+    [CommonData(setname='NMC', ndata=204, commondataproc='DIS_NCE', nkin=3, nsys=16)]
+    """
+    level0_commondata_instances_wc = []
+
+    #==== Load validphys.coredata.CommonData instance with cuts ====#
+    
+    for j, dataset in enumerate(data.datasets):
+        if dataset.cuts is None:
+            commondata_wc = dataset.commondata.load_commondata_instance()
+        else:
+            cuts = dataset.cuts.load()
+            commondata_wc = dataset.commondata.load_commondata_instance().with_cuts(cuts)
+        
+    
+        #== Generate a new CommonData instance with central value given by Level 0 data generated with fakepdf ==#
+        from validphys.covmats import dataset_t0_predictions
+        t0_prediction = dataset_t0_predictions(dataset=dataset, t0set=fakepdf)  # N.B. cuts already applied to th. pred.
+        level0_commondata_instances_wc.append(commondata_wc.with_central_value(t0_prediction))
+
+    return level0_commondata_instances_wc
 
+
+
+def make_level1_data(data,commondata_wc,level0_commondata_wc,filterseed):
```

Comment:
I think we should have a vp provider that does `commondata_wc`  provider, so that this function can be used as an action.

Reply:
I added a `commondata_wc` function in `results.py`. Also, in the new commits the `make_level0_data` has been renamed `level0_commondata_wc`